### ISSUE TITLE
Annotated PDFs Being Overwritten by WeBeep Sync

### ISSUE THREAD

**Issue by mattiacolombomc** [toto04/webeep-sync#104]
While using WeBeep Sync, I've encountered a situation that leads to loss of annotations on PDF files. Here's the sequence of events:

1. WeBeep Sync downloads a set of PDF slides.
2. I promptly open the file from the current directory with a PDF editor, especially when I'm in a hurry.
3. I annotate and take notes on the PDF throughout the lecture.
4. After the lecture, I save the annotated file.
5. Post-saving, WeBeep Sync detects that a file is absent and re-downloads the original file, inadvertently overwriting and erasing all my annotations.

It would be immensely helpful if we could find a workaround for this situation.

**Comment by toto04** [comment]
yep this is definitely the biggest issue with this app right now, im yet to find the perfect solution to this problem, as it is right now the app detects a discrepancy in the "last modified" timestamp and overwrites the file, this is essential to keep files updated if the professor happens to update them, if I were to modify this behavior so that the app only overwrites files with older timestamps than the ones on webeep that still wouldn't be a bullet-proof solution, as if the professor edits the file it would still get overwritten on the user's pc.

I'm not really sure how I would go about detecting user editing of the file, and even if I did I'm also not sure about what's the best way of handling it on the fs, I could either keep an original copy, marking it in some way (like adding "[ORIGINAL]" at the end of the file path) but then I would have to keep track of such a change, or just stop keeping the file synced all together and not download it even if it gets updated on webeep, but then of course new changes of the file from the professor wouldn't be downloaded and that wouldn't be great either.

I am definitely open to suggestions lol
